plugins/woocommerce/src/Internal/Logging/RemoteLogger.php

```diff
@@ -16,10 +17,202 @@
  * @since 9.2.0
  * @package WooCommerce\Classes
  */
-class RemoteLogger {
+class RemoteLogger extends \WC_Log_Handler {
+	const LOG_ENDPOINT                = 'https://public-api.wordpress.com/rest/v1.1/logstash';
+	const THROTTLE_TRANSIENT          = 'woocommerce_remote_logging_throttle';
+	const THROTTLE_LIMIT              = 10; // Maximum number of requests in the interval.
+	const THROTTLE_INTERVAL           = 300; // Throttle interval in seconds (5 minutes).
 	const WC_LATEST_VERSION_TRANSIENT = 'latest_woocommerce_version';
 	const FETCH_LATEST_VERSION_RETRY  = 'fetch_latest_woocommerce_version_retry';
 
+	/**
+	 * The logger instance.
+	 *
+	 * @var \WC_Logger_Interface|null
+	 */
+	private $local_logger;
+
+	/**
+	 * Remote logger constructor.
+	 *
+	 * @internal
+	 * @param \WC_Logger_Interface|null $logger Logger instance.
+	 */
+	public function __construct( \WC_Logger_Interface $logger = null ) {
+		if ( null === $logger ) {
+			$this->local_logger = wc_get_logger();
+		}
+		$this->local_logger = $logger;
+	}
+
+	/**
+	 * Handle a log entry.
+	 *
+	 * @param int    $timestamp Log timestamp.
+	 * @param string $level emergency|alert|critical|error|warning|notice|info|debug.
+	 * @param string $message Log message.
+	 * @param array  $context Optional. Additional information for log handlers.
+	 *
+	 * @return bool False if value was not handled and true if value was handled.
+	 */
+	public function handle( $timestamp, $level, $message, $context ) {
+		// Do nothing here.
```

Comment:
I'm a bit confused why we do nothing here. In the interface, it looks like `handle` is intended to do the job that `log` currently has.

To comply with interface, perhaps `log` method should be renamed to `handle` and take the first $timestamp param (but do nothing with the timestamp) 🤔

Reply:
We don't actually use it as wc logger handler in the code (at least for our first implementation). As mentioned in previous comment, I extended WC_Log_Handler just because of reusing `get_backtrace` method from `WC_Log_Handler`.

But that sounds like a good idea to rename `log` method to `handle` so make it less confusing. I will update the PR with this change. Thanks for the suggestion!

Reply:
Renamed it here e1989eaa3a59c3f1112782c6a7aed4157298711c

Reply:
I've given it a bit more thought. 

What do you think about extracting `get_backtrace` to a trait (perhaps LogUtils?) and removing the interface? I think we should not try to implement an interface just to use a utility function from an abstract class. It can confuse developers in the future.

Reply:
@moon0326  As discussed with Barry above, we will likely add a remote logger as a WC logger handler. So, I think it's good to keep the interface for now.  Let me know if you have different thoughts on this. Thanks! 🙏

Reply:
Yep: if we see potential for this to expand (in the future) beyond just fatal/critical errors, I think that makes a lot of sense and—especially if we eventually integrate it in the same way as other logging implementations—it would provide a better degree of consistency (3PDs etc seeking to control logging through existing hooks don't have to figure out how this other system works).

That said, we certainly have a considerable amount of freedom to test different strategies and change them later (because much of the work lives in the internal namespace), so also don't treat my thoughts as a blocker of any kind.

Reply:
Thanks @chihsuan @barryhughes 👍 

Makes sense 🚀